src/assisted_test_infra/test_infra/helper_classes/day2_cluster.py

```diff
@@ -4,155 +4,106 @@
 import subprocess
 import time
 import uuid
-from abc import ABC
 from typing import Any
 
 import waiting
+from junit_report import JunitTestCase
 
 import consts
-from assisted_test_infra.test_infra import BaseInfraEnvConfig, utils
+from assisted_test_infra.test_infra import BaseInfraEnvConfig, ClusterName, utils
+from assisted_test_infra.test_infra.helper_classes.base_cluster import BaseCluster
 from assisted_test_infra.test_infra.controllers import NodeController
-from assisted_test_infra.test_infra.helper_classes.config import BaseDay2ClusterConfig
+from assisted_test_infra.test_infra.helper_classes.cluster import Cluster
+from assisted_test_infra.test_infra.helper_classes.config.base_day2_cluster_config import BaseDay2ClusterConfig
 from assisted_test_infra.test_infra.tools import static_network
 from assisted_test_infra.test_infra.utils.waiting import wait_till_all_hosts_are_in_status
 from service_client import log
 from service_client.assisted_service_api import InventoryClient
 
 
-class Day2Cluster(ABC):
-    def __init__(
-        self, api_client: InventoryClient, config: BaseDay2ClusterConfig, infra_env_config: BaseInfraEnvConfig
-    ):
-        self.config = config
-        self.api_client = api_client
-        self._infra_env_config = infra_env_config
-
-    def prepare_for_installation(self, iso_download_path: str = None):
-        utils.recreate_folder(consts.IMAGE_FOLDER, force_recreate=False)
+class Day2Cluster(BaseCluster):
+    _config: BaseDay2ClusterConfig
 
-        cluster = self.api_client.cluster_get(cluster_id=self.config.day1_cluster_id)
-        self.config.day1_cluster_name = cluster.name
-        openshift_version = cluster.openshift_version
-        api_vip_dnsname = "api." + self.config.day1_cluster_name + "." + cluster.base_dns_domain
-        api_vip_ip = cluster.api_vip
+    def __init__(self, config: BaseDay2ClusterConfig, infra_env_config: BaseInfraEnvConfig, cluster: Cluster):
```

Comment:
just for extra-explicitness 
```suggestion
    def __init__(self, config: BaseDay2ClusterConfig, infra_env_config: BaseInfraEnvConfig, day1_cluster: Cluster):
```

Reply:
it's actually a fixture, changing the name of the parameter breaks the tests :face_exhaling: